cmd/signimage/signimage.go

```diff
@@ -3,21 +3,141 @@ package main
 import (
 	"archive/tar"
 	"bytes"
+	"encoding/json"
 	"flag"
 	"fmt"
-	"github.com/docker/cli/cli/config"
-	dockertypes "github.com/docker/cli/cli/config/types"
 	"github.com/go-logr/logr"
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/kubernetes-sigs/kernel-module-management/internal/registry"
 	"io"
+	"io/fs"
 	"k8s.io/klog/v2/klogr"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strings"
 )
 
+// configDir should contain all the secrets available as individual files named for their keys
+// so we need to search through for the secrets we need for our pull and push repos.
+// and if we find nothingi default anonymous auth
+// we're (trying to be) tolerant in our inputs, precise in our outputs, and opaque in our comments. Its the UNIX way!
+type repoAuth struct {
+	pullRepo  string
+	pushRepo  string
+	PullAuth  authn.Authenticator
+	PushAuth  authn.Authenticator
+	configDir string
+}
+
+func NewRepoAuth(configDir string, pullRepo string, pushRepo string) *repoAuth {
+	if pushRepo == "" {
+		pushRepo = pullRepo
+	}
+	r := &repoAuth{
+		pullRepo:  pullRepo,
+		pushRepo:  pushRepo,
+		PullAuth:  nil,
+		PushAuth:  nil,
+		configDir: filepath.Clean(configDir),
+	}
+	r.PopulateAuthFromFileList()
+	if r.PullAuth == nil {
+		r.PullAuth = authn.Anonymous
+		logger.Info("no secrets for pull found, default to Anonymous")
+	}
+	if r.PushAuth == nil {
+		r.PushAuth = authn.Anonymous
+		logger.Info("no secrets for push found, default to Anonymous")
+	}
+	return r
+}
+
+func (r *repoAuth) PopulateAuthFromFileList() {
+	// not giving any secrets should short cuircuit the whole process
+	if r.configDir == "" {
+		return
+	}
```

Comment:
How about setting the anonymous authn as default? Then we can remove some ifs in the calling function.
```suggestion
	r.PullAuth = authn.Anonymous
	r.PushAuth = authn.Anonymous

	// not giving any secrets should short cuircuit the whole process
	if r.configDir == "" {
		return
	}
```

Reply:
because I wanted to be able to log that its using anonymous auth, so when it fails to push or pull you it gives you a clue that its because you didn't give it the right secrets. It might be that's what you want so its a message not an error, but the user needs to know so we would still need the if its just a question of `if nil` or `if authn.Anonymous` and the former feels more like its catching an error to me  

signimage is deliberately quite aggressive at logging what its doing so when it goes wrong its (hopefully) obvious what needs fixing